src/routers/controllers/editMemberRoleController.ts

```diff
@@ -0,0 +1,86 @@
+import { Request, Response } from "express";
+import logger from "../../lib/Logger";
+import * as constants from "../../lib/constants";
+import { getTranslationsForView } from "../../lib/utils/translationUtils";
+import { UserRoleChangeData, ViewData } from "../../types/utilTypes";
+import { getExtraData, getLoggedUserAcspMembership, setExtraData } from "../../lib/utils/sessionUtils";
+import { Membership } from "../../types/membership";
+import { isValidRole } from "../../lib/validation/user.role.validation";
+import { addErrorToViewData } from "../../lib/utils/viewUtils";
+import { FormInputNames } from "../../lib/validation/add.user.validation";
+import { sanitizeUrl } from "@braintree/sanitize-url";
+import { UserRole } from "private-api-sdk-node/dist/services/acsp-manage-users/types";
+
+export const editMemberRoleControllerGet = async (req: Request, res: Response): Promise<void> => {
+    const viewData = await getViewData(req);
+    if (viewData.isTheOnlyOwner) {
+        const userRoleChangeData = getUserRoleChangeData(req, viewData);
+        setExtraData(req.session, constants.USER_ROLE_CHANGE_DATA, userRoleChangeData);
+        res.redirect(constants.STOP_PAGE_ADD_ACCOUNT_OWNER_FULL_URL);
+    } else {
+        res.render(constants.EDIT_MEMBER_ROLE_PAGE, viewData);
+    }
+};
+
+export const editMemberRoleControllerPost = async (req: Request, res: Response): Promise<void> => {
```

Comment:
do we need to check all these here? as these will be checked on backend. like if its a valid user.
if the user is allowed to change role and all the below logic.

I though, we will just be collecting information and passing on the detail to back end.

However, we return standard error message from api, rather than the error which is required.